java/integTest/src/test/java/glide/SharedCommandTests.java

```diff
@@ -897,6 +897,53 @@ public void sismember(BaseClient client) {
         assertTrue(executionException.getCause() instanceof RequestException);
     }
 
+    @SneakyThrows
+    @ParameterizedTest(autoCloseArguments = false)
+    @MethodSource("getClients")
+    public void sdiffstore(BaseClient client) {
+        String key1 = "{key}-1-" + UUID.randomUUID();
+        String key2 = "{key}-2-" + UUID.randomUUID();
+        String key3 = "{key}-3-" + UUID.randomUUID();
+        String key4 = "{key}-4-" + UUID.randomUUID();
+        String key5 = "{key}-5-" + UUID.randomUUID();
+
+        assertEquals(3, client.sadd(key1, new String[] {"a", "b", "c"}).get());
+        assertEquals(3, client.sadd(key2, new String[] {"c", "d", "e"}).get());
+        assertEquals(3, client.sadd(key4, new String[] {"e", "f", "g"}).get());
+
+        // create new
+        assertEquals(2, client.sdiffstore(key3, new String[] {key1, key2}).get());
+        assertEquals(Set.of("a", "b"), client.smembers(key3).get());
+
+        // overwrite existing set
+        assertEquals(2, client.sdiffstore(key2, new String[] {key3, key2}).get());
+        assertEquals(Set.of("a", "b"), client.smembers(key2).get());
+
+        // overwrite source
+        assertEquals(3, client.sdiffstore(key1, new String[] {key1, key4}).get());
+        assertEquals(Set.of("a", "b", "c"), client.smembers(key1).get());
+
+        // overwrite source
+        assertEquals(3, client.sdiffstore(key1, new String[] {key1}).get());
+        assertEquals(Set.of("a", "b", "c"), client.smembers(key1).get());
+
+        // source key exists, but it is not a set
```

Comment:
we need an example that diffs a empty key (& assumes it's an empty set). 
You could use key5 before setting to a string.